docs/architecture_guide.md

```diff
@@ -34,10 +45,86 @@ An issue with ResNet-style shortcuts comes up when there is a dimensionality cha
 
 We have found that using 8-bit weights and activations in high-precision shortcuts is sufficient to absorb the output from convolutions; increasing the bit-width beyond that yields no further accuracy improvement. Additionally, we recommend to use as many shortcuts as you can: for example, in ResNet-style architectures it helps to bypass every single convolutional layer, instead of every two convolutional layers.
 
+An example of a convolutional block with shortcut in Larq could look something like this:
+
+```python
+def residual_block(x, filters=None, strides=1):
+    """Convolutional block with shortcut connection. The shortcut uses average
+     pooling for resolution changes and a 1x1 convolution if (and only if)
+     there is a change in number of filters.
+
+    Args:
+      x: input tensor with high precision activations
+      filters: number of output filters (if None, use number of input filters)
+      strides: An integer or tuple/list of 2 integers, specifying the strides
+        of the convolution along the height and width. Can be a single integer
+        to specify the same value for all spatial dimensions
+
+    Returns:
+      Tensor with high-precision activations
+    """
+    # compute dimensions
+    in_filters = x.get_shape().as_list()[-1]
+    out_filters = filters or in_filters
+
+    # create shortcut that retains the high-precision information
+    shortcut = x
+    if strides != 1:
+        shortcut = tf.keras.layers.AvgPool2D(
+            pool_size=strides,
+            strides=strides,
+            padding="same",
+            )(shortcut)
+    if in_filters != out_filters:
+        shortcut = tf.keras.layers.Conv2D(
+            out_filters,
+            1,
+            kernel_initializer="glorot_normal",
+            use_bias=False,
+        )(shortcut)
+        shortcut = tf.keras.layers.BatchNormalization(momentum=0.9)(shortcut)
```

Comment:
I personally would remove the handling for changing the dimension in the, to make this example easier to understand. Since there are many ways how you could handle dimension changes I don't think we should add this here as an example.